setuptools/_core_metadata.py

```diff
@@ -175,7 +176,7 @@ def write_field(key, value):
         if attr_val is not None:
             write_field(field, attr_val)
 
-    license = self.get_license()
+    license = self.license_expression or self.get_license()
```

Comment:
So this mean that we need a follow-up PR for the `License-Expression`, right?

Reply:
Correct. This PR was designed so that it itself didn't require a metadata_version bump. That could / should happen at a later point once all pieces are in place. I.e. setuptools has adopted version `2.3`, the license files are stored in the correct folder and license expression parsing is handled correctly.